controllers/random_node_selector.go

```diff
@@ -13,7 +15,7 @@ type RandomNodeSelector struct {
 }
 
 func NewRandomNodeSelector(asg *autoscaling.Group, ruObj *upgrademgrv1alpha1.RollingUpgrade) *RandomNodeSelector {
-	maxUnavailable := getMaxUnavailable(ruObj.Spec.Strategy, len(asg.Instances))
+	maxUnavailable := getMaxUnavailable(ruObj.Spec.Strategy, int(aws.Int64Value(asg.DesiredCapacity)))
```

Comment:
Isn't this the same? I guess desired is a more stable number?

Reply:
`len(asg.Instance)` will also consider the standBy instances. This will adversely impact the % caluclation. For % calculation we should only include the Inservice instances, right?